### DISCUSSION TITLE
E3SM port machine to Ubuntu:  pioc_support.c:2018:5: error: conflicting typec_writemap

### DISCUSSION THREAD

**Discussion by daofeng2007** [E3SM-Project/E3SM#7768] [E3SM model help]
I am trying to port E3SM machine to a Ubuntu server.
System: Ubuntu 18.04
E3SM version: maint1.0.0
Package versions:
zlib-1.2.12
hdf5-1_12_2
netcdf-c-4.9.0
netcdf-fortran-4.6.0
pnetcdf-1.12.3
mpich-4.0.2
gfortran-10.3.0
gcc-10.3.0

./case.build

Finished creating component namelists
Building gptl with output to file /data/taoliu/taoliufile/my_e3sm/myoutput/t6/bld/gptl.bldlog.230106-094227
Calling /data/taoliu/taoliufile/my_e3sm/E3SM/cime/src/build_scripts/buildlib.gptl
Component gptl build complete with 3 warnings
Building mct with output to file /data/taoliu/taoliufile/my_e3sm/myoutput/t6/bld/mct.bldlog.230106-094227
Calling /data/taoliu/taoliufile/my_e3sm/E3SM/cime/src/build_scripts/buildlib.mct
Component mct build complete with 42 warnings
Building pio with output to file /data/taoliu/taoliufile/my_e3sm/myoutput/t6/bld/pio.bldlog.230106-094227
Calling /data/taoliu/taoliufile/my_e3sm/E3SM/cime/src/build_scripts/buildlib.pio
ERROR: /data/taoliu/taoliufile/my_e3sm/E3SM/cime/src/build_scripts/buildlib.pio FAILED, cat /data/taoliu/taoli3sm/myoutput/t6/bld/pio.bldlog.230106-094227

cat /data/taoliu/taoli3sm/myoutput/t6/bld/pio.bldlog.230106-094227
/data/taoliu/taoliufile/my_e3sm/E3SM/externals/scorpio/src/clib/pioc_support.c:2018:5: error: conflicting typec_writemap

[config_machines.xml.txt](https://github.com/E3SM-Project/E3SM/files/10362218/config_machines.xml.txt)
[config_compilers.xml.txt](https://github.com/E3SM-Project/E3SM/files/10362219/config_compilers.xml.txt)
[pio.bldlog.230106-094227.txt](https://github.com/E3SM-Project/E3SM/files/10362221/pio.bldlog.230106-094227.txt)

**Comment by whannah1** [comment]
@daofeng2007 despite config_compilers still being in the repo we don't use it anymore in favor of a CMAKE build system, which might be the source of your issue. I wish I could point you to a place to get more info on that, but I don't have such a link. You might have better luck asking about this problem on the "e3sm-help" slack channel. We generally don't use github issues for addressing non-supported machine issues.
